arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -678,6 +678,15 @@ void RocksDBEngine::start() {
     _options.max_open_files = -1;
   }
 
+  auto shaFileManager = std::make_shared<RocksDBShaFileManager>(_path);
+  if (_createShaFiles) {
+
+    // Register checksum factory
+    _options.file_checksum_gen_factory = std::make_shared<RocksDBSha256ChecksumFactory>(shaFileManager);
+    shaFileManager->checkMissingShaFiles();
+    _options.listeners.push_back(shaFileManager);
+  }
```

Comment:
`shaFileManager` seems to be a local variable and not used outside of the `if` block:
```suggestion
  if (_createShaFiles) {
    auto shaFileManager = std::make_shared<RocksDBShaFileManager>(_path);
    // Register checksum factory
    _options.file_checksum_gen_factory = std::make_shared<RocksDBSha256ChecksumFactory>(shaFileManager);
    // Do an initial check in the database directory for missing SHA checksum files
    shaFileManager->checkMissingShaFiles();
    _options.listeners.push_back(std::move(shaFileManager));
  }
```